crates/starknet-artifacts/src/lib.rs

```diff
@@ -0,0 +1,551 @@
+use anyhow::{anyhow, Context, Result};
+use camino::{Utf8Path, Utf8PathBuf};
+use scarb_metadata::{CompilationUnitMetadata, Metadata, MetadataCommand, PackageId};
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::fs;
+use test_collector::LinkedLibrary;
+
+#[derive(Deserialize, Debug, PartialEq, Clone)]
+struct StarknetArtifacts {
+    version: u32,
+    contracts: Vec<StarknetContract>,
+}
+
+#[allow(dead_code)]
+#[derive(Deserialize, Debug, PartialEq, Clone)]
+struct StarknetContract {
+    id: String,
+    package_name: String,
+    contract_name: String,
+    artifacts: StarknetContractArtifactPaths,
+}
+
+#[allow(dead_code)]
+#[derive(Deserialize, Debug, PartialEq, Clone)]
+struct StarknetContractArtifactPaths {
+    sierra: Utf8PathBuf,
+    casm: Utf8PathBuf,
+}
+
+/// Contains compiled Starknet artifacts
+#[derive(Debug, PartialEq, Clone)]
+pub struct StarknetContractArtifacts {
+    /// Compiled sierra code
+    pub sierra: String,
+    /// Compiled casm code
+    pub casm: String,
+}
+
+/// Get deserialized contents of `starknet_artifacts.json` file generated by Scarb
+///
+/// # Arguments
+///
+/// * `path` - A path to `starknet_artifacts.json` file.
+fn artifacts_for_package(path: &Utf8Path) -> Result<StarknetArtifacts> {
+    let starknet_artifacts =
+        fs::read_to_string(path).with_context(|| format!("Failed to read {path:?} contents"))?;
+    let starknet_artifacts: StarknetArtifacts =
+        serde_json::from_str(starknet_artifacts.as_str())
+            .with_context(|| format!("Failed to parse {path:?} contents. Make sure you have enabled sierra and casm code generation in Scarb.toml"))?;
```

Comment:
I know cast requires both, but to be double sure is this true for forge as well? @piotmag769

Reply:
Yup, we need both (sierra is used for class hash computation)